tests/test_nacl_loss.py

```diff
@@ -0,0 +1,228 @@
+                            [0.2703, 0.7722, 0.1611, 0.6431],
+                            [0.8051, 0.6596, 0.4121, 0.1125],
+                            [0.5283, 0.6746, 0.5528, 0.7913],
+                        ],
+                    ]
+                ]
+            ),
+            "targets": torch.tensor([[[1, 1, 1, 1], [1, 1, 1, 0], [0, 0, 0, 0], [0, 0, 0, 0]]]),
+        },
+        1.1820,
+    ],
+    [
+        {"classes": 3, "dim": 2, "kernel_ops": "gaussian"},
+        {
+            "inputs": torch.tensor(
+                [
+                    [
+                        [
+                            [0.1498, 0.1158, 0.3996, 0.3730],
+                            [0.2155, 0.1585, 0.8541, 0.8579],
+                            [0.6640, 0.2424, 0.0774, 0.0324],
+                            [0.0580, 0.2180, 0.3447, 0.8722],
+                        ],
+                        [
+                            [0.3908, 0.9366, 0.1779, 0.1003],
+                            [0.9630, 0.6118, 0.4405, 0.7916],
+                            [0.5782, 0.9515, 0.4088, 0.3946],
+                            [0.7860, 0.3910, 0.0324, 0.9568],
+                        ],
+                        [
+                            [0.0759, 0.0238, 0.5570, 0.1691],
+                            [0.2703, 0.7722, 0.1611, 0.6431],
+                            [0.8051, 0.6596, 0.4121, 0.1125],
+                            [0.5283, 0.6746, 0.5528, 0.7913],
+                        ],
+                    ]
+                ]
+            ),
+            "targets": torch.tensor([[[1, 1, 1, 1], [1, 1, 1, 0], [0, 0, 0, 0], [0, 0, 0, 0]]]),
+        },
+        1.1850,
+    ],
+    [
+        {"classes": 3, "dim": 2, "kernel_ops": "gaussian"},
+        {
+            "inputs": torch.tensor(
+                [
+                    [
+                        [
+                            [0.0411, 0.3386, 0.8352, 0.2741],
+                            [0.4821, 0.0519, 0.2561, 0.9391],
+                            [0.5954, 0.4184, 0.9160, 0.7977],
+                            [0.0588, 0.9156, 0.1307, 0.9914],
+                        ],
+                        [
+                            [0.8481, 0.0892, 0.2044, 0.8470],
+                            [0.3558, 0.3569, 0.0979, 0.4491],
+                            [0.0876, 0.0929, 0.4040, 0.8384],
+                            [0.5313, 0.3927, 0.4165, 0.1107],
+                        ],
+                        [
+                            [0.7993, 0.6938, 0.3151, 0.8728],
+                            [0.7332, 0.4111, 0.3862, 0.9988],
+                            [0.2622, 0.5002, 0.1905, 0.1644],
+                            [0.6354, 0.0047, 0.1649, 0.7112],
+                        ],
+                    ]
+                ]
+            ),
+            "targets": torch.tensor([[[1, 2, 0, 1], [0, 2, 1, 2], [0, 0, 2, 1], [1, 1, 1, 2]]]),
+        },
+        1.0375,
+    ],
```

Comment:
This test appears to test the same thing as the previous? Does this add anything?

Reply:
Sure, got it. I will remove one of the case, it is just different values for the input and target tensors.